layers/cmd/ingest_csv.go

```diff
@@ -195,8 +198,210 @@ func (ci *CSVIngester) Run(pipeline *pipeline.PipelineContext) error {
 	return nil
 }
 
+type CSVJoinIngester struct {
+	BaseIngestParams
+	StartRow    int    `json:"startRow" yaml:"startRow"`
+	EndRow      int    `json:"endRow" yaml:"endRow"`
+	HeaderRow   int    `json:"headerRow" yaml:"headerRow"`
+	ID          string `json:"id" yaml:"id"`
+	Delimiter   string `json:"delimiter" yaml:"delimiter"`
+	ColumnRange uint   `json:"columnRange" yaml:"columnRange"`
+	initialized bool
+	ingester    []*ls.Ingester
+	entities    []CSVJoinConfig
+}
+
+type CSVJoinConfig struct {
+	VariantID        string
+	StartCol, EndCol int
+	IDCols           []int
+}
+
+type joinData struct {
+	CSVJoinConfig
+	data []string
+	id   string
+}
+
+func (cji *CSVJoinIngester) Run(pipeline *pipeline.PipelineContext) error {
+	var layer *ls.Layer
+	var err error
+	var schLoader ls.SchemaLoader
+	if !cji.initialized {
+		schLoader, err = LoadBundle(pipeline.Context, cji.Bundle)
+		if err != nil {
+			return err
+		}
+		cji.initialized = true
+	}
+
+	for i := 0; ; i++ {
+		entryInfo, stream, err := pipeline.NextInput()
+		if err != nil {
+			return err
+		}
+		if stream == nil {
+			break
+		}
+		pipeline.Context.GetLogger().Debug(map[string]interface{}{"csvingest": "start new stream"})
+		reader := csv.NewReader(stream)
+		if cji.Schema != "" {
+			return errors.New("Unexpected schema")
+		}
+		layer, err = schLoader.LoadSchema(cji.entities[i].VariantID)
+		if err != nil {
+			return err
+		}
+		pipeline.Properties[fmt.Sprintf("layer: %s", cji.entities[i].VariantID)] = layer
+		cji.ingester[i] = &ls.Ingester{Schema: layer}
+		parser := csvingest.Parser{
```

Comment:
You are also going to need a different parser for each ingester. So, have a separate loop here going through the entities and creating those. i is not the entity index here,